File: app/src/main/java/com/BlizzardArmory/URLConstants.java
package com.BlizzardArmory;

import com.BlizzardArmory.ui.MainActivity;

/**
 * The type Url constants.
 */
public class URLConstants {
    /**
     * The constant loading.
     */
//Loading boolean
    public static boolean loading = false;

    /**
     * The constant paypalURL.
     */
//Paypal
    public final static String paypalURL = "https://paypal.me/astpierredev";

    /**
     * The constant CALLBACK_URL.
     */
//Callback URL
    public final static String CALLBACK_URL = "https://alexchaoss.github.io/BnetAuthorize";

    /**
     * The constant BASE_URL_USER_INFO.
     */
//Account information
    public final static String BASE_URL_USER_INFO = "https://zone.battle.net/";
    /**
     * The constant BASE_URL_CN_USER_INFO.
     */
    public final static String BASE_URL_CN_USER_INFO = "https://www.battlenet.com.cn";

    /**
     * The constant BASE_URL_API.
     */
//Base URL for APIs
    public final static String BASE_URL_API = "https://zone.api.blizzard.com/";
    /**
     * The constant BASE_URL_CN_API.
     */
    public final static String BASE_URL_CN_API = "https://gateway.battlenet.com.cn/";

    /**
     * The constant ACCESS_TOKEN_QUERY.
     */
//Access Token Query
    public final static String ACCESS_TOKEN_QUERY = "access_token=";

    /**
     * The constant NOT_FOUND_URL_AVATAR.
     */
//URLs for WoW images
    public final static String NOT_FOUND_URL_AVATAR = "?alt=/wow/static/images/2d/avatar/";
    /**
     * The constant MEDIA_QUERY.
     */
    public final static String MEDIA_QUERY = "/profile/wow/character/realm/charactername/character-media?namespace=profile-us&locale=en_US&access_token=";

    /**
     * The constant D3_ICON_ITEMS.
     */
//URLS for D3 characters
    public final static String D3_ICON_ITEMS = "http://media.blizzard.com/d3/icons/items/large/icon.png";
    /**
     * The constant D3_ICON_SKILLS.
     */
    public final static String D3_ICON_SKILLS = "http://media.blizzard.com/d3/icons/skills/64/url.png";

    /**
     * The constant OW_PROFILE.
     */
//URL Overwwatch
    public final static String OW_PROFILE = "https://ow-api.com/v1/stats/:platform/:region/:battletag/complete";

    /**
     * Gets base ur lfor user information.
     *
     * @return the base ur lfor user information
     */
    public static String getBaseURLforUserInformation() {
        if (MainActivity.selectedRegion.toLowerCase().equals("cn")) {
            return URLConstants.BASE_URL_CN_USER_INFO;
        } else {
            return URLConstants.BASE_URL_USER_INFO.replace("zone", MainActivity.selectedRegion.toLowerCase());
        }
    }

    /**
     * Gets base ur lfor api.
     *
     * @param region the region
     * @return the base ur lfor api
     */
    public static String getBaseURLforAPI(String region) {
        if (region == null || "".equalsIgnoreCase(region)) {
            if (MainActivity.selectedRegion.equals("cn")) {
                return URLConstants.BASE_URL_CN_API;
            } else {
                return URLConstants.BASE_URL_API.replace("zone", MainActivity.selectedRegion.toLowerCase());
            }
        } else {
            return URLConstants.BASE_URL_API.replace("zone", region.toLowerCase());
        }
    }

    /**
     * Gets region.
     *
     * @return the region
     */
    public static String getRegion() {
        return MainActivity.selectedRegion.toLowerCase();
    }

    /**
     * Gets ow profile.
     *
     * @param username the username
     * @param platform the platform
     * @return the ow profile
     */
    public static String getOWProfile(String username, String platform) {
        String url;
        if (platform.equalsIgnoreCase("PC")) {
            url = OW_PROFILE.replace(":battletag", username.replace("#", "-"));
            url = url.replace(":platform", "pc");
            if (MainActivity.selectedRegion.toLowerCase().equals("cn") || MainActivity.selectedRegion.toLowerCase().equals("tw")) {
                url = url.replace(":region", "asia");
            } else {
                url = url.replace(":region", MainActivity.selectedRegion.toLowerCase());
            }
        } else {
            url = OW_PROFILE.replace(":battletag", username);
            url = url.replace(":platform", platform);
            url = url.replace(":region", "global");
        }
        return url;
    }
}

File: app/src/main/java/com/BlizzardArmory/network/URLConstants.kt
package com.BlizzardArmory.network

import com.BlizzardArmory.ui.main.MainActivity

/**
 * The type Url constants.
 */
object URLConstants {
    /**
     * The constant loading.
     */
    @JvmField
    var loading = false
    var HEROKU_AUTHENTICATE = "https://blizzardgamesprofiles.herokuapp.com/"

    /**
     * The constant paypalURL.
     */
    const val PAYPAL_URL = "https://paypal.me/astpierredev"

    /**
     * The constant CALLBACK_URL.
     */
    const val CALLBACK_URL = "https://alexchaoss.github.io/BnetAuthorize"
    const val LOGOUT_URL = "https://battle.net/login/logout"

    /**
     * The constant BASE_URL_USER_INFO.
     */
    const val BASE_URL_USER_INFO = "https://zone.battle.net/"

    /**
     * The constant BASE_URL_CN_USER_INFO.
     */
    const val BASE_URL_CN_USER_INFO = "https://www.battlenet.com.cn"

    /**
     * The constant NOT_FOUND_URL_AVATAR.
     */
    const val NOT_FOUND_URL_AVATAR = "?alt=/wow/static/images/2d/avatar/"

    /**
     * The constant D3_ICON_ITEMS.
     */
    const val D3_ICON_ITEMS = "http://media.blizzard.com/d3/icons/items/large/icon.png"

    /**
     * D3 images
     */
    const val D3_ASSETS = "https://alexchaoss.github.io/BnetAuthorize/img/d3/image.png"

    /**
     * The constant D3_ICON_SKILLS.
     */
    const val D3_ICON_SKILLS = "http://media.blizzard.com/d3/icons/skills/64/url.png"

    /**
     * The constant OW_PROFILE.
     */
    const val OW_PROFILE = "https://ow-api.com/v1/stats/:platform/:region/:battletag/complete"

    /**
     * Overwatch images
     */
    const val OW_ASSETS = "https://alexchaoss.github.io/BnetAuthorize/img/ow/image.png"

    /**
     * SC2 images
     */
    const val SC2_ASSETS = "https://alexchaoss.github.io/BnetAuthorize/img/sc2/image.jpg"

    /**
     * WoW images
     */
    const val WOW_ASSETS = "https://alexchaoss.github.io/BnetAuthorize/img/wow/image.jpg"

    /**
     * Gets base url for user information.
     *
     * @return the base url for user information
     */
    @JvmStatic
    val baseURLforUserInformation: String
        get() = if (MainActivity.selectedRegion.equals("cn", ignoreCase = true)) {
            BASE_URL_CN_USER_INFO
        } else {
            BASE_URL_USER_INFO.replace("zone", MainActivity.selectedRegion.toLowerCase())
        }

    /**
     * Gets region.
     *
     * @return the region
     */
    val region: String
        get() = MainActivity.selectedRegion.toLowerCase()

    /**
     * Gets ow profile.
     *
     * @param username the username
     * @param platform the platform
     * @return the ow profile
     */
    @JvmStatic
    fun getOWProfile(username: String, platform: String): String {
        var url: String
        if (platform.equals("PC", ignoreCase = true)) {
            url = OW_PROFILE.replace(":battletag", username.replace("#", "-"))
            url = url.replace(":platform", "pc")
            url = if (MainActivity.selectedRegion.equals("cn", ignoreCase = true) || MainActivity.selectedRegion.equals("tw", ignoreCase = true)) {
                url.replace(":region", "asia")
            } else {
                url.replace(":region", MainActivity.selectedRegion.toLowerCase())
            }
        } else {
            url = OW_PROFILE.replace(":battletag", username)
            url = url.replace(":platform", platform)
            url = url.replace(":region", "global")
        }
        return url
    }

    @JvmStatic
    fun getOWPortraitImage(character: String): String {
        return OW_ASSETS.replace("image", character + "_portrait")
    }

    fun getOWIconImage(character: String): String {
        return OW_ASSETS.replace("image", character + "_icon")
    }

    fun getD3Asset(name: String): String {
        return D3_ASSETS.replace("image", name)
    }

    fun getSC2Asset(name: String): String {
        return SC2_ASSETS.replace("image", name)
    }

    fun getWoWAsset(name: String): String {
        return WOW_ASSETS.replace("image", name)
    }

    fun selectAchievementsURLFromLocale(locale: String): String {
        return "https://blizzard-armory-server.herokuapp.com/achievements/$locale"
    }

    fun selectAchievementCategoriesURLFromLocale(locale: String): String {
        return "https://blizzard-armory-server.herokuapp.com/categories/$locale"
    }

    fun getTalentsIcons(playableClassId: Int, locale: String): String {
        return "https://blizzard-armory-server.herokuapp.com/talents/$playableClassId/$locale"
    }

    fun getCovenantClassSpells(playableClassId: Int, locale: String): String {
        return "https://blizzard-armory-server.herokuapp.com/covenant/class/$playableClassId/$locale"
    }

    fun getCovenantSpells(covenantId: Int, locale: String): String {
        return "https://blizzard-armory-server.herokuapp.com/covenant/$covenantId/$locale"
    }

    fun getTechTalents(soulbindId: Int, locale: String): String {
        return "https://blizzard-armory-server.herokuapp.com/tech_talents/$soulbindId/$locale"
    }

    fun getReputations(locale: String): String {
        return "https://blizzard-armory-server.herokuapp.com/reputations/$locale"
    }

}